pkg/source/mtadapter/controller.go

```diff
@@ -64,10 +65,28 @@ func NewController(ctx context.Context, adapter adapter.Adapter) *controller.Imp
 	kafkasourceInformer := kakfasourceinformer.Get(ctx)
 	kafkasourceInformer.Informer().AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
-			AddFunc:    impl.Enqueue,
-			UpdateFunc: controller.PassNew(impl.Enqueue),
+			AddFunc: impl.Enqueue,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				object, err := kmeta.DeletionHandlingAccessor(newObj)
+
+				if err != nil {
+					r.logger.Errorw("UpdateFunc", zap.Error(err))
+					return
+				}
+
+				if !object.GetDeletionTimestamp().IsZero() {
+					// Stopping the adapter might take a while, so run this in a goroutine to
+					// not block the informer
+					go func() {
+						r.mtadapter.Remove(object.GetName(), object.GetNamespace())
```

Comment:
Sorry for not understanding maybe - why are we doing this "remove" in the UpdateFunc and not DeleteFunc?

Reply:
`remove` is called in both places. Actually `DeleteFunc` may not be needed anymore, but does hurt to keep it.

Reply:
Got it. Ok.